e2e/benchmark/benchmark_extractor.go

```diff
@@ -0,0 +1,154 @@
+// Copyright © 2024 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package benchmark
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/fbiville/markdown-table-formatter/pkg/markdown"
+
+	apitypes "github.com/openclarity/openclarity/api/types"
+)
+
+const tableTitle = "# 🚀 Benchmark results"
+
+type BenchmarkExtractor interface {
+	ExportAsMarkdown(file string) error
+	header() []string
+	rows() [][]string
+}
+
+type familyStat struct {
+	name  string
+	stats *[]apitypes.AssetScanInputScanStats
+}
+
+type benchmark struct {
+	familyStats        []familyStat
+	generalStats       *apitypes.AssetScanGeneralStats
+	totalFindingsCount int
+}
+
+func NewBenchmarkExtractor(scanStats *apitypes.AssetScanStats) (BenchmarkExtractor, error) {
+	b := &benchmark{
+		generalStats: scanStats.General,
+	}
+	for _, family := range getFamiliesStats(scanStats) {
+		if family.stats != nil {
+			b.familyStats = append(b.familyStats, family)
+		}
+	}
+
+	return b, nil
+}
+
+func (b *benchmark) header() []string {
+	return []string{"Family/Scanner", "Start time", "End time", "Findings", "Total time"}
+}
+
+func (b *benchmark) rows() [][]string {
+	var familyPerScannerRows [][]string //nolint:prealloc
+	var summaryRows [][]string          //nolint:prealloc
+
+	for _, family := range b.familyStats {
+		earliestStartTime := time.Time{}
+		latestEndTime := time.Time{}
+		totalFindingsCount := 0
+		for _, scanner := range *family.stats {
+			if earliestStartTime.IsZero() || scanner.ScanTime.StartTime.Before(earliestStartTime) {
+				earliestStartTime = *scanner.ScanTime.StartTime
+			}
+			if latestEndTime.IsZero() || scanner.ScanTime.EndTime.After(latestEndTime) {
+				latestEndTime = *scanner.ScanTime.EndTime
+			}
+			totalFindingsCount += *scanner.FindingsCount
```

Comment:
this value is not true since we merge scanner findings from each scanner into family findings. some findings can be rediscovered across scanners, so `sum(all family scanner total findings) != total family fidings`. this value can be read from the scan result.

We can handle this in another PR since the difference is not really big